src/solvers/smt/smt_conv.cpp

```diff
@@ -718,6 +728,19 @@ smt_astt smt_convt::convert_ast(const expr2tc &expr)
     a = args[0]->project(this, 0);
     break;
   }
+  case expr2t::pointer_capability_id:
+  {
+    assert(config.ansi_c.cheri);
+    const pointer_capability2t &obj = to_pointer_capability2t(expr);
+    // Potentially walk through some typecasts
```

Comment:
Why?

Reply:
For the same reason the cases
- expr2t::pointer_offset_id, and
- expr2t::pointer_object_id
do. :) Basically, the expression can contain typecasts which under these kinds of expressions are ignored.